apps/omg_watcher_info/test/omg_watcher_info/block_application_consumer_test.exs

```diff
@@ -0,0 +1,80 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherInfo.BlockApplicationConsumerTest do
+  use ExUnitFixtures
+  use ExUnit.Case, async: true
+
+  alias OMG.TestHelper
+  alias OMG.Watcher.BlockGetter.BlockApplication
+  alias OMG.WatcherInfo.BlockApplicationConsumer
+  alias OMG.WatcherInfo.DB
+  alias OMG.WatcherInfo.DB.PendingBlock
+
+  @eth OMG.Eth.zero_address()
+
+  setup tags do
+    {:ok, pid} =
+      BlockApplicationConsumer.start_link(
+        bus_module: __MODULE__.FakeBus,
+        name: BlockApplicationConsumerTest
+      )
+
+    Map.put(tags, :pid, pid)
+  end
+
+  describe "handle_info/2" do
+    @tag fixtures: [:phoenix_ecto_sandbox]
+    test "inserts the given block application into pending block", %{pid: pid} do
+      alice = TestHelper.generate_entity()
+      bob = TestHelper.generate_entity()
+
+      tx_1 = TestHelper.create_recovered([{1, 0, 0, alice}], @eth, [{bob, 300}])
+      tx_2 = TestHelper.create_recovered([{1, 0, 0, alice}], @eth, [{bob, 500}])
+
```

Comment:
are these perhaps unnecessarily complex? the data that block_application_consumer can accept is really arbitrary. and the test should only prove that :term_to_binary and :binary_to_term does what it needs to do